vspace/src/main/webapp/WEB-INF/views/exhibition/downloads/downloadList.html

```diff
@@ -0,0 +1,214 @@
+            }
+        });
+    }
+    function triggerExhibitionSnapshotCreation(){ 
+        $.ajax({
+            type : "POST",
+            url : "[(@{'/staff/exhibit/snapshot?'+ ${_csrf.parameterName}+'='+${_csrf.token} })] ",
+            cache : false,
+            
+            contentType : false,
+            processData : false,
+            
+            success : function(data) {
+            	pollForExhibitionDownload(data.id, data.folderName); 		
+            },
+            error : function(data) {					
+            	var alert = $('<div class="alert alert-danger alert-dismissible fade show" role="alert"><p>Snapshot could not be downloaded</p><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button></div>');
+            	$('.errorAdd').append(alert);
+            	
+            	var errorAlert = document.getElementById('downloadError');
+            	errorAlert.style.visibility="visible";
+            }
+        });
+    }
+
+    function pollForExhibitionDownload(id, folderName) {	
+    	var alert = $('<div class="alert alert-info alert-dismissible fade show" role="alert"><p>Snapshot '+folderName+' creation is in progress. Please wait. </p><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button></div>');
+    	$('.successAlert').append(alert);
+    	const poll = (id, folderName) => {
+            maxAttempts = 10;
+            interval= 3000;
+            let attempts = 0;
+    
+            const executePoll = async (resolve, reject) => {
+                const  url = "[(@{'/staff/exhibit/download/checkStatus/'})]"+id+"?folderName="+folderName+','+ [[${_csrf.parameterName}]]+'='+[[${_csrf.token}]];			  
+                const result = await fetch(url);
+                attempts++;
+                var text = await result.text();
+                if ( text == "true") {
+                    var alert = $('<div class="alert alert-success alert-dismissible fade show" role="alert"><p>Snapshot '+folderName+' is ready. Please click to download. <a href=[[@{/staff/exhibit/download/'+id+'?folderName='+folderName+'}]] > <span class="btn primary-btn btn-sm">Download</span></a> </p><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button></div>');
+                    $('.successAlert').empty();
+                    $('.successAlert').append(alert); 
+                    return resolve(result);
+                } else if (maxAttempts && attempts === maxAttempts) {
+                    var alert = $('<div class="alert alert-danger alert-dismissible fade show" role="alert"><p>Snapshot '+folderName+' could not be created. Please try again later</p><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button></div>');
+                    $('.errorAdd').append(alert);
```

Comment:
that's not necessarily true, is it? it might just take longer. it's fine to set a limit on the polling but instead of saying there was an error, it should say the user should check back at a certain url.

Reply:
changed it